syntax = "proto2";

package skal;

message Alarm {
    enum Severity {
        NOTICE = 0;
        WARNING = 1;
        ERROR = 2;
    }
    optional string name = 1;
    optional Severity severity = 2;
    optional bool is_on = 3;
    optional bool auto_off = 4;
    optional string note = 5;
    optional string origin = 6;
    optional int64 timestamp = 7;
}

message IntField {
    optional string name = 1;
    optional sint64 value = 2;
}

message DoubleField {
    optional string name = 1;
    optional double value = 2;
}

message StringField {
    optional string name = 1;
    optional string value = 2;
}

message MiniblobField {
    optional string name = 1;
    optional bytes value = 2;
}

message BlobField {
    optional string name = 1;
    optional string allocator = 2;
    optional string id = 3;
}

message Msg {
    optional uint32 version = 1;
    optional int64 timestamp = 2;
    optional string sender = 3;
    optional string recipient = 4;
    optional string action = 5;
    optional int32 ttl = 6;
    optional uint32 flags = 7;
    optional uint32 iflags = 8;
    repeated Alarm alarms = 9;
    repeated IntField int_fields = 10;
    repeated DoubleField double_fields = 11;
    repeated StringField string_fields = 12;
    repeated MiniblobField miniblob_fields = 13;
    repeated BlobField blob_fields = 14;
}
